crates/cheatcodes/spec/src/cheatcode.rs

```diff
@@ -120,6 +120,12 @@ pub enum Group {
     ///
     /// Safety: safe.
     Utilities,
+    /// Utility cheatcodes to check forge execution context.
+    ///
+    /// Examples: `isTestContext`, `isScriptBroadcastContext`.
+    ///
+    /// Safety: safe.
+    Context,
```

Comment:
I think this can just be one function that takes in an enum rather than one function per variant. Then the group is also unnecessary. CC @mattsse @klkvr

Reply:
you mean have one function returning enum instead of several `is*` methods? I think the motivation for this impl was discussed here https://github.com/foundry-rs/foundry/issues/2900#issuecomment-1687334918

we can probably create one function which would return something like 
```solidity
enum ForgeContext {
    Test,
    Script,
    Coverage,
    Snapshot,
    DryRun,
    Broadcast
}

struct ForgeContextData {
    ForgeContext context;
    bool isTest;
    bool isScript;
}

```

Reply:
> you mean have one function returning enum instead of several `is*` methods? I think the motivation for this impl was discussed here [#2900 (comment)](https://github.com/foundry-rs/foundry/issues/2900#issuecomment-1687334918)
> 

yep, indeed, that was the motivation cc @mds1

Reply:
Another approach would be a single `isContext(ForgeContext ctx)` method where:
```solidity
struct ForgeContext {
  ScriptDryRun,
  ScriptBroadcast,
  ScriptGroup, // equal to ScriptDryRun || ScriptBroadcast
  Coverage,
  Snapshot,
  Test,
  TestGroup, // equal to Coverage || Snapshot || Test
  // etc...
}
```

Ultimately these all feel similarly clean and usable so I'm fairly indifferent. I do think the nested `ForgeContextData` struct above is my least favorite since it's a little clunkier